app/jobs/data_warehouse/stale_data_check_job.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module DataWarehouse
+  class StaleDataCheckJob < BaseJob
+    REPORT_NAME = 'idp_max_ids'
+
+    def perform(timestamp)
+      data = fetch_table_max_ids_and_counts(timestamp)
+      upload_to_s3(data, timestamp)
+    end
+
+    def fetch_table_max_ids_and_counts(timestamp)
+      max_ids_and_counts(timestamp)
+    end
+
+    private
+
+    def max_ids_and_counts(timestamp)
+      active_tables = {}
+      ActiveRecord::Base.connection.tables.each do |table|
+        if table_has_id_column?(table)
+          active_tables[table] = fetch_max_id_and_count(table, timestamp)
+        end
+      end
+
+      active_tables
+    end
+
+    def table_has_id_column?(table)
+      ActiveRecord::Base.connection.columns(table).map(&:name).include?('id')
+    end
+
+    def fetch_max_id_and_count(table, timestamp)
+      query = <<-SQL
+        SELECT COALESCE(MAX(id), 0) AS max_id, COUNT(*) AS row_count
```

Comment:
`COUNT(*)` is going to be pretty slow in production

Reply:
We did check this in int on the largest tables we could find... Im not worried about it being fast for our purposes but rather the potential implications for Postgres impact... Do you think it would have worse impact than many of our reports?

Reply:
We should use the `transaction_with_timeout` helper which will
1. remove the query timeout so we can actually get a result
2. use the read replica so it won't slow down writes in prod

Reply:
`int` is probably not going to reflect performance in prod very well. Read replica means it's unlikely to impact normal operations though.

Reply:
It would be orders of magnitude faster without the count since it seems like the `created_at` aspect could be dropped and then you're only selecting a single indexed column. If it's about stale checks, max ID might be sufficient? Not a dealbreaker either way.

Reply:
So the reason we are including the created at is because we cannot guarantee that table state at the time of this check running will be the same as the state when the data is ingested from idp side to the data warehouse. The created_at threshold was to try hopefully force an equal look-back to prevent any false positives

We wanted count in there as a troubleshooting tool and later as another potential piece to monitor to track accumulating deltas and make sure we have the correct deletions/modifications when they occur in the idp. From a alerting standpoint we are initially only going to alert off a delta max_id so we are on the same page there.

Great questions!

We will implement read replica changes and query timeout modifications. Thanks for the suggestion!